Code/Framework/AzToolsFramework/AzToolsFramework/AssetBundle/AssetBundleComponent.cpp

```diff
@@ -626,7 +627,11 @@ namespace AzToolsFramework
     
     bool AssetBundleComponent::InjectFile(const AZStd::string& filePath, const AZStd::string& sourcePak)
     {
-        return InjectFile(filePath, sourcePak, "");
+        // When no working directory is specified, assume that the file being injected goes into the root of the archive.
+        // The filePath should be an absolute path, making the workingDirectory be the path leading up to the file.
+        AZ::IO::Path fullFilePath{ filePath, AZ::IO::PosixPathSeparator };
+        AZStd::string workingDir{ fullFilePath.RemoveFilename().Native() };
+        return InjectFile(filePath, sourcePak, workingDir.c_str());
```

Comment:
`RemoveFilename` should be used here.
It is not meant to remove a directory from path. It is meant to empty out the last path segment

What should be used here is `ParentPath`

```suggestion
        AZ::IO::PathView fullFilePath{ filePath, AZ::IO::PosixPathSeparator };
        AZ::IO::Path workingDir{ fullFilePath.ParentPath() };
        return InjectFile(filePath, sourcePath.workingDir.c_str());
```

Reply:
Ok, that's better.